java/src/com.zeroc.ice/src/main/java/com/zeroc/Ice/UserException.java

```diff
@@ -54,18 +60,27 @@ public void _read(InputStream is) {
     }
 
     /**
+     * Returns {@code true} if this exception uses classes, {@code false} otherwise.
+     *
+     * @return {@code true} if this exception uses classes, {@code false} otherwise
```

Comment:
I know that the method is literally named `usesClasses`, but I think it's a pretty lame name.
What we really care about are "Slice classes". In Java, all exceptions "use" classes, since all exceptions _are_ classes.
```suggestion
     * Returns {@code true} if this exception holds classes, {@code false} otherwise.
     *
     * @return {@code true} if this exception holds classes, {@code false} otherwise
```